shared/utils/index.ts

```diff
@@ -45,3 +45,8 @@ export const getCategoryTiles = ({
       isExternalUrl: externalOfferUrl !== undefined,
     })
   })
+
+export const hasProp = <P extends PropertyKey, O extends { [p in P]: unknown }>(
+  obj: O,
+  p: P,
+): obj is O & { [p in P]: NonNullable<unknown> } => obj[p] !== undefined && obj[p] !== null
```

Comment:
🙃  Can we call them object and prop? I don't think we really use abbreviations anywhere else, and that would make it marginally more readable, at least to me.